spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/config/SpringDocAutoConfigurationTests.java

```diff
@@ -0,0 +1,195 @@
+					verifyCustomizerHasIgnorePatterns(context, "/swagger-ui/**",
+							"/v3/**",
+							"/swagger-ui.html",
+							"/v3/api-docs/swagger-config",
+							"https://validator.swagger.io/validator",
+							"/swagger-ui/oauth2-redirect.html",
+							"/webjars",
+							"/webjars/**"
+					);
+				});
+	}
+
+	@Test
+	void enabledWithCustomSpringDocSettings() {
+		contextRunnerWithSpringDocEnabled()
+				.withPropertyValues(
+						"springdoc.api-docs.path=/v4/foo/api-docs",
+						"springdoc.swagger-ui.config-url=/v4/bar/swagger-config")
+				.run((context) -> {
+					assertThat(context)
+							.hasSingleBean(SpringDocAutoConfiguration.class)
+							.hasBean("springDocWebSecurityCustomizer")
+							.hasBean("springDocJsonDecodeFilterRegistration");
+					verfiyFilterHasUrlPatterns(context,"/v4/foo", "/v4/foo/*", "/v4/bar", "/v4/bar/*");
+					verifyCustomizerHasIgnorePatterns(context, "/swagger-ui/**",
+							"/v4/foo/**",
+							"/swagger-ui.html",
+							"/v4/bar/swagger-config",
+							"https://validator.swagger.io/validator",
+							"/swagger-ui/oauth2-redirect.html",
+							"/webjars",
+							"/webjars/**"
+					);
+				});
+	}
+
+	private void verfiyFilterHasUrlPatterns(AssertableWebApplicationContext context, String... expected) {
+		FilterRegistrationBean<SpringDocJsonDecodeFilter> filterRegistration =
+				context.getBean("springDocJsonDecodeFilterRegistration", FilterRegistrationBean.class);
+		assertThat(filterRegistration.getUrlPatterns()).containsExactlyInAnyOrder(expected);
+	}
+
+	private void verifyCustomizerHasIgnorePatterns(AssertableWebApplicationContext context, String... expected) {
+		WebSecurityCustomizer customizer = context.getBean("springDocWebSecurityCustomizer", WebSecurityCustomizer.class);
+		WebSecurity webSecurity = mock(WebSecurity.class, Answers.RETURNS_DEEP_STUBS);
+		customizer.customize(webSecurity);
+		ArgumentCaptor<String> antMatchersCaptor = ArgumentCaptor.forClass(String.class);
+		verify(webSecurity.ignoring()).antMatchers(antMatchersCaptor.capture());
+		assertThat(antMatchersCaptor.getAllValues()).containsExactly(expected);
+	}
+
+	@Test
+	void defaultsAreInSyncWithSpringdoc() {
+		contextRunnerWithSpringDocEnabled()
+				.run((context) -> {
+					SpringDocConfigProperties springDocConfigProps = context.getBean(SpringDocConfigProperties.class);
+					SwaggerUiConfigProperties swaggerUiConfigProps = context.getBean(SwaggerUiConfigProperties.class);
+					assertThat(springDocConfigProps.getApiDocs().getPath()).isEqualTo(Constants.DEFAULT_API_DOCS_URL);
+					assertThat(swaggerUiConfigProps.getPath()).isEqualTo(Constants.DEFAULT_SWAGGER_UI_PATH);
+					assertThat(swaggerUiConfigProps.getConfigUrl()).isEqualTo(
+							Constants.DEFAULT_API_DOCS_URL + AntPathMatcher.DEFAULT_PATH_SEPARATOR + Constants.SWAGGGER_CONFIG_FILE);
+					assertThat(swaggerUiConfigProps.getValidatorUrl()).isEqualTo("https://validator.swagger.io/validator");
```

Comment:
Just a question: Is there no Constant for this, too?

Reply:
Good eye! Yes there is https://github.com/springdoc/springdoc-openapi/blob/378edcf142d9b2c5af637339b926f98687e55650/springdoc-openapi-common/src/main/java/org/springdoc/core/Constants.java#L61 and I attempted to use that. However, you will see it references https://github.com/springdoc/springdoc-openapi/blob/378edcf142d9b2c5af637339b926f98687e55650/springdoc-openapi-common/src/main/java/org/springdoc/core/Constants.java#L51 which contains SpEL which will fail a basic equality check here. The other alternative would be to evaluate the constant but it seemed more hassle than it was worth so I went w/ a pragmatic hybrid approach.

Reply:
Yes right! 👍